src/main/java/io/fabric8/elasticsearch/plugin/KibanaUserReindexFilter.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.elasticsearch.plugin;
+
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.commons.lang.StringUtils;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.netty.handler.codec.http.HttpRequest;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.http.netty.NettyHttpRequest;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestFilter;
+import org.elasticsearch.rest.RestFilterChain;
+import org.elasticsearch.rest.RestRequest;
+
+public class KibanaUserReindexFilter extends RestFilter {
+
+	private static final String DEFAULT_AUTH_PROXY_HEADER = "X-Proxy-Remote-User";
+	private static final String SEARCHGUARD_AUTHENTICATION_PROXY_HEADER = "searchguard.authentication.proxy.header";
```

Comment:
If these are not already in the ConfigurationSettings interface of our plugin, they should be